DNN Platform/Library/UI/Skins/SkinScope.cs

```diff
@@ -1,13 +1,10 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information
-
-namespace DotNetNuke.UI.Skins
-{
-    public enum SkinScope
-    {
-        All = 0,
-        Host = 1,
-        Site = 2,
-    }
-}
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System.Runtime.CompilerServices;
+
+using DotNetNuke.UI.Skins;
+
+// The enum 'SkinScope' has been moved to DotNetNuke.Abstractions
+[assembly: TypeForwardedTo(typeof(SkinScope))]
```

Comment:
How does this know to forwared to DotNetNuke.Abstractions ? Is this right or we need to fully qualify that type?

Reply:
Because DotNetNuke.Library has a reference to DotNetNuke.Abstractions. The `typeof` is the type in the assembly DotNetNuke.Abstractions, so the compiler knows which assembly the type has moved to.

It's also not possible to add a fully qualifier in this attribute, [since it only accepts `System.Type`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.typeforwardedtoattribute.-ctor?view=netframework-4.7.2#system-runtime-compilerservices-typeforwardedtoattribute-ctor(system-type)).

More info about TypeForwardTo: https://learn.microsoft.com/en-us/dotnet/standard/assembly/type-forwarding

Reply:
What would happen if SkinScope type existsted in multiple other referenced assemblies ? How would it know which one to forward to ?

Reply:
I think it depends on the namespace.

If the class/enum exists in multiple assemblies with the same namespace, you would get a ambiguous reference. In this case you can't even compile the library.

Since the enum has been moved to Abstractions, with the same namespace, it'll be fine.

Reply:
I haven't used type forwarding, but AFAIK it should work fine in this scenario. I guess the only concern here is that we don't get to change the namespace to `DotNetNuke.Abstractions.Skins`, it has to remain `DotNetNuke.UI.Skins`. Is it worth introducing the conversion code between the two separate `enum` types so that we can have cleaner organization in the long-term?

Reply:
Should we rename the enum in the abstractions assembly?
Otherwise, you'll have two `SkinType`'s; one deprecated and one that you should use. You'll have to make sure to get the right namespace or fill in the full enum name (`DotNetNuke.Abstractions.Skins.SkinType`)

Reply:
The other enums that we've moved have not been renamed, we just require using the right namespace. It is certainly messier in the short term to do it that way, I'm not certain if it's worth it in the long term or not.